pilot/pkg/model/authorization.go

```diff
@@ -30,162 +29,168 @@ type RolesAndBindings struct {
 	// ServiceRoles in the same namespace.
 	Roles []Config
 
-	// Maps from ServiceRole name to its associated ServiceRoleBindings.
-	RoleNameToBindings map[string][]*rbacproto.ServiceRoleBinding
+	// ServiceRoleBindings indexed by its associated ServiceRole's name.
+	Bindings map[string][]*rbacproto.ServiceRoleBinding
 }
 
-// AuthorizationPolicyConfig stores the AuthorizationPolicy and its name.
-type AuthorizationPolicyConfig struct {
-	Name string
+// AuthorizationPolicies organizes authorization policies by namespace.
+type AuthorizationPolicies struct {
+	// Maps from namespace to RolesAndBindings.
+	namespaceToV1Policies map[string]*RolesAndBindings
+
+	// The mesh global RbacConfig.
+	rbacConfig *rbacproto.RbacConfig
```

Comment:
Quick question. Is the goal of the refactor to make this code reusable for v1 and v2? How do we plan to handle the structures like ServiceRoleBinding or RbacConfig, which are not in v2 policy?

Reply:
There are 2 goals here: 1) make it easier for adding authz v2 support later. 2) Improve the current unit testing. We won't change the ServiceRoleBinding or RbacConfig in v2 support, we will just add a new field called `namespaceToV2Policies` and support v1 and v2 policy at the same time.

Reply:
Not 100% what is your plan, but I feel it is better to have separate model for each version (and then separate generator). It's cleaner as they are well separated and later you can just delete the whole v1 folder when the time come. Just my 2c.

Reply:
Discussed offline, for the model in pilot/pkg/model we can either share it in the same one or refactor it into two, it doesn't matter too much as it's the collection of the raw authz CR.

The real authZ model is in pilot/pkg/security/authz/model/model.go that supports perfectly both v1 and v2 policy.